src/types/objects.rs

```diff
@@ -0,0 +1,85 @@
+//! Implementations for the types describing DIF object files.
+
+use super::{
+    AllObjectCandidates, Arch, CodeId, CompleteObjectInfo, DebugId, ObjectCandidate,
+    ObjectFeatures, ObjectFileStatus, ObjectUseInfo, RawObjectInfo,
+};
+use crate::sources::{SourceId, SourceLocation};
+
+impl Default for ObjectUseInfo {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+impl ObjectUseInfo {
+    pub fn is_none(&self) -> bool {
+        matches!(self, Self::None)
+    }
+}
+
+impl From<Vec<ObjectCandidate>> for AllObjectCandidates {
+    fn from(mut source: Vec<ObjectCandidate>) -> Self {
+        source.sort_by_key(|candidate| (candidate.source.clone(), candidate.location.clone()));
+        Self(source)
+    }
+}
+
+impl AllObjectCandidates {
+    /// Sets the [`ObjectCandidate::debug`] field for the specified DIF object.
+    ///
+    /// You can only request symcaches from a DIF object that was already in the metadata
+    /// candidate list, therefore if the candidate is missing it is treated as an error.
+    pub fn set_debug(&mut self, source: SourceId, location: SourceLocation, info: ObjectUseInfo) {
+        match self
+            .0
+            .binary_search_by_key(&(source, location), |candidate| {
+                (candidate.source.clone(), candidate.location.clone())
```

Comment:
I think the same applies here. Maybe factoring out the sorting function would be a good idea, as that would also format this a lot nicer.

Reply:
Not sure what your suggestion is here.  There is no cached variant, which is probably because this never needs to compute the same key twice.  As this is currently only in the `From` and here I don't think it's worth further factoring out.  Once this grows more methods, like `.set_unwind()` that'll probably change though at which point I think it makes more sense to do that.